src/components/CartButtons.js

```diff
@@ -1,35 +1,14 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
 import { FaShoppingCart, FaUserMinus, FaUserPlus } from 'react-icons/fa';
 import { Link } from 'react-router-dom';
 import styled from 'styled-components';
 import { useProductsContext } from '../context/products_context';
-import axios, { AxiosError } from 'axios';
-
-const rootUrl = 'https://ecommerce-6kwa.onrender.com';
+import { useUserContext } from '../context/user_context';
 
 //vart-btn-wrapper- global class- see in Navbar.js it is display none on a default screen, nested class in NavContainer = styled...
 const CartButtons = () => {
   const { closeSidebar } = useProductsContext(); // extracting the closeSidebar function from the returned object from useProductsContext() and assigning it to a variable named closeSidebar.
-  const [currentUser, setCurrentUser] = useState(null);
-
-  useEffect(() => {
-    async function fetchData() {
-      const url = `${rootUrl}/api/v1/users/showMe`;
-      axios
-        .get(url, { withCredentials: true })
-        .then((response) => {
-          console.log(response);
-          setCurrentUser(response.data.user);
-        })
-        .catch((error) => {
-          const errorPayload =
-            error instanceof AxiosError ? error.response.data : error;
-          console.error(errorPayload);
-        });
-    }
-    fetchData();
-    // By using empty array [], for the "dependencies" argument of useEffect, it tells React to run this useEffect hook only *once*, the first time this component/context is rendered
-  }, []);
```

Comment:
Instead of doing this in multiple components, we just do it once in the user\_context.js component